content/client-lib-development-guide/features.textile

```diff
@@ -375,15 +378,20 @@ h3(#realtime-channels). Channels
 * @(RTL2)@ EventEmitter and states:
-** @(RTL2a)@ Implements @EventEmitter@ and emits events for state changes @INITIALIZED@, @ATTACHING@, @ATTACHED@, @DETACHING@, @DETACHED@, @FAILED@
+** @(RTL2a)@ Implements @EventEmitter@ and emits events for state changes @INITIALIZED@, @ATTACHING@, @ATTACHED@, @DETACHING@, @DETACHED@, @SUSPENDED@ and @FAILED@
 ** @(RTL2b)@ @Channel#state@ attribute is the current state of the channel
+** @(RTL2d)@ A @ChannelStateChange@ object is emitted as the first argument for every connection state change
+** @(RTL2e)@ Additionally, a @ChannelStateChange@ can be emitted that contains a @reason@ which contains an @ErrorInfo@ object with details of the error that has occurred for the @Channel@
+** @(RTL2f)@ When a channel @ATTACHED@ @ProtocolMessage@ is received, the @ProtocolMessage@ may contain a bit flag with value 4 indicating that the channel has been resumed. The @ChannelStateChange@ will then be emitted for the @ATTACHED@ state and will contain a @resumed@ boolean attribute with value @true@ if the bit flag value 4 was included. When @resumed@ is @true@, this indicates that the channel attach resumed the channel state from an existing connection and there has been no loss of message continuity. In all other cases, @resumed@ is false. A test should exist to ensure that @resumed@ is always false when a channel first becomes @ATTACHED@, it is @true@ when the channel is @ATTACHED@ following a successful "connection recovery":#RTN16, and is @false@ when the channel is @ATTACHED@ following a failed "connection recovery":#RTN16
 ** @(RTL2c)@ Additionally, an @ERROR@ event is emitted that contains an @ErrorInfo@ object with details on an error that has occurred for the @Channel@
 * @(RTL3)@ Connection state change side effects:
 ** @(RTL3a)@ If the connection state changes to @FAILED@ then an @ATTACHING@ or @ATTACHED@ channel state will transition to @FAILED@, set the @Channel#errorReason@ and emit the error event
-** @(RTL3b)@ If the connection state changes to @CLOSED@ or @SUSPENDED@ then an @ATTACHING@ or @ATTACHED@ channel state will transition to @DETACHED@
-* @(RTL11)@ If a channel enters the @DETACHED@ or @FAILED@ state, then all messages that are still queued for send on that channel should be deleted from the queue triggering a failure for the publish or presence methods invoked for those messages
+** @(RTL3b)@ If the connection state changes to @CLOSED@ then an @ATTACHING@ or @ATTACHED@ channel state will transition to @DETACHED@
+** @(RTL3c)@ If the connection state changes to @SUSPENDED@ then an @ATTACHING@ or @ATTACHED@ channel state will transition to @SUSPENDED@
+** @(RTL3d)@ If the connection state changes to @CONNECTED@ then a @SUSPENDED@ channel state will transition to @ATTACHING@. If any of the channels automatically attached fail, then the error message received from Ably should be emitted as an @ERROR@ event on the @Channel@ with Ably error code @91200@
```

Comment:
> `If any of the channels automatically attached fail, then the error message received from Ably should be emitted as an @ERROR@ event on the @Channel@`

Why do we need this? If the channel fails (eg realtime rejects the autoattach), it will emit a `failed` event with the reason realtime gives for the failure. Why emit an `error` as well?

(We discussed emitting an error on the channel it in the other issue, but as I understood it, that was for a failed presence autoreenter (which otherwise wouldn't result in an event emitted, unlike a channel failure))